### PR TITLE
[wasm] Add task marshaling tests.

### PR DESCRIPTION
**Pull request by maraf**
Add various tests for marshaling task to JavaScript promise.

- Sync and async.
- Generic and non-generic.
- Successful and failed.

These tests replace those added in the dotnet/runtime#47640.

@pavelsavara @kg  Do you have some ideas for additional scenarios to test?

### CODE REVIEW

**Comment by kg** [comment]
Looks good. I'd add a test for ```ValueTask``` and a test for ```ValueTask<int>``` - I believe they'll fail at present but it's worth documenting that through tests that assert they fail, since we might get requests to implement it later on.

File: src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/MarshalTests.cs [hunk 1]
```diff
@@ -899,5 +901,94 @@ public static void InvokeJSNotInGlobalScope()
...
+            var exception = await Assert.ThrowsAsync<JSException>(() => MarshalTask("AsynchronousValueTask"));
+            Assert.StartsWith("Error: no idea on how to unbox value types", exception.Message);
+        }
+
+        [Fact]
+        [Trait("Category","Marek")]
+        public static async Task MarshalAsynchronousValueTaskIntDoesNotWorkYet()
+        {
+            var exception = await Assert.ThrowsAsync<JSException>(() => MarshalTask("AsynchronousValueTaskInt", "7"));
+            Assert.StartsWith("Error: no idea on how to unbox value types", exception.Message);
```
Comment (pavelsavara) [inline src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/MarshalTests.cs:991]: Let's make assertion as if it worked and mark the test with [ActiveIssue] and create that issue

Comment (maraf) [reply src/libraries/System.Private.Runtime.InteropServices.JavaScript/tests/System/Runtime/InteropServices/JavaScript/MarshalTests.cs]: Updated.

**Comment by maraf** [comment]
I have added the same bunch of tests for ValueTasks as for Tasks, as I'm expecting the marshaling implementation will be different. Let me know if it's ok or we don't need all of them.

**Review by pavelsavara — APPROVED** [review]

**Pull request merged**